frontend/app/components/wp-table/timeline/global-elements/wp-timeline-relations.directive.ts

```diff
@@ -103,18 +101,21 @@ export class WorkPackageTableTimelineRelations {
    * Ensure visible relations (through table.rows) are loaded automatically.
    */
   private requireVisibleRelations() {
+    const whenTimelineIsVisibleNext = () => {
+      return this.states.table.timelineVisible.values$()
+        .filter((timelineState:WorkPackageTableTimelineState) => timelineState.isVisible)
+        .take(1);
+    }
 
-    // Observe the rows and request relations if changed
-    // AND timeline is visible.
-    Observable.combineLatest(
-      this.states.table.timelineVisible.values$().takeUntil(scopeDestroyed$(this.$scope)),
-      this.states.table.rendered.values$().takeUntil(scopeDestroyed$(this.$scope))
-    )
-      .filter(([timelineState, rendered]) => timelineState.isVisible)
-      .subscribe(() => {
-        this.workPackageIdOrder = this.getVisibleWorkPackageOrder();
-        this.wpRelations.requireInvolved(this.workPackageIdOrder);
+    // Observe the rows and remember if changed
+    // so we can request them as soon as the timeline is visible
+    this.states.table.rows.values$().takeUntil(scopeDestroyed$(this.$scope))
+      .subscribe((rows) => {
+
+      whenTimelineIsVisibleNext().subscribe(() => {
+        this.wpRelations.requireInvolved(rows.map(el => el.id));
```

Comment:
As this is a reference to `states.table.rows.value`, it does not matter whether its kept in another closure.